pkg/inspector/image-inspector.go

```diff
@@ -166,27 +166,40 @@ func aggregateBytesAndReport(bytesChan chan int) {
 // decodeDockerPullMessages will parse the docker pull messages received
 // from reader and will push the difference of bytes downloaded to
 // bytesChan. After reader is closed it will close bytesChan and exit.
-func decodeDockerPullMessages(bytesChan chan int, reader io.Reader) {
+func decodeDockerPullMessages(parsedErrors chan bool, bytesChan chan int, reader io.Reader) {
 	type progressDetailType struct {
 		Current, Total int
 	}
 	type pullMessage struct {
 		Status, Id     string
 		ProgressDetail progressDetailType
+		Error          string
 	}
 	defer func() { close(bytesChan) }()           // Closing the channel to end the other routine
+	defer func() { close(parsedErrors) }()        // Close channel
 	layersBytesDownloaded := make(map[string]int) // bytes downloaded per layer
 	dec := json.NewDecoder(reader)                // decoder for the json messages
+
+	var startedDownloading = false
 	for {
 		var v pullMessage
 		if err := dec.Decode(&v); err != nil {
 			if err != io.ErrClosedPipe {
 				log.Printf("Error decoding json: %v", err)
+			} else {
+				parsedErrors <- false
 			}
 			break
 		}
 		// decoding
+		if v.Error != "" {
```

Comment:
what writes to the channel in the case of no decode error and no `v.Error`?  Is that going to block the call below of `if parsedError := <-parsedErrors; parsedError {` forever?  Lots going on here, I couldn't quite tell if that's a valid use case (will pull this PR today and see if I can reproduce anything)